particle-iot/particle-cli
# Use control requests for updating the bootloader

## Description

This PR updates the `update` command so that it can flash certain modules of the target release, such as the bootloader module, using control requests. Consequently, the `update` command no longer requires the target device to be in DFU mode and instead automatically switches it to an appropriate state before flashing.

Other changes in this PR:

- Ensure there's a single place where the list of supported platforms is defined (see platform.js). Previously, different commands/subsystems of the CLI supported different sets of platforms due to the definitions from device-constants being filtered differently in those subsystems. Consequently, the remaining support for the following platforms was removed in this PR: asom, xsom, raspi, oak, duo, bluz.
- Change the order in which the firmware modules are updated so that the bootloader is flashed first.
- Use the device ID to identify the device when flashing it via `dfu-util`.

## How to Test

- Validate that the `update` and `flash` commands work as expected.
- Run unit tests.
- Run end-to-end tests.

## Related Issues / Discussions

- [sc-104580](https://app.shortcut.com/particle/story/104580/particle-update-command-cannot-be-used-to-update-bootloader-on-p2)

## Completeness
